Library/Formula/gdb.rb

```diff
@@ -11,6 +11,19 @@ def message; <<-EOS.undent
   end
 end
 
+class UniversalBrewedPython3 < Requirement
+  satisfy { archs_for_command("python3").universal? }
+
+  def message; <<-EOS.undent
+    A build of GDB using a brewed Python3 was requested, but Python3 is not
+    a universal build.
+
+    GDB requires Python3 to be built as a universal binary or it will fail
+    if attempting to debug a 32-bit binary on a 64-bit host.
```

Comment:
This applies with normal GDB too so I don't think it should be a requirement but instead a caveat.